ee/tables/windowsupdatetable/cache_windows.go

```diff
@@ -0,0 +1,125 @@
+//go:build windows
+// +build windows
+
+package windowsupdatetable
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/observability"
+)
+
+type (
+	// windowsUpdatesCacher queries for fresh Windows updates data every `cacheInterval`,
+	// and stores it in the `cacheStore`.
+	windowsUpdatesCacher struct {
+		cacheStore    types.GetterSetter
+		cacheInterval time.Duration
+		queryCancel   context.CancelFunc
+		slogger       *slog.Logger
+		interrupt     chan struct{}
+		interrupted   atomic.Bool
+	}
+
+	// cachedQueryResults stores the results of querying the Windows Update Agent API;
+	// it includes a timestamp so we know how fresh this data is.
+	cachedQueryResults struct {
+		QueryTime time.Time     `json:"timestamp"`
+		Results   *QueryResults `json:"results"`
+	}
+)
+
+func NewWindowsUpdatesCacher(cacheStore types.GetterSetter, cacheInterval time.Duration, slogger *slog.Logger) *windowsUpdatesCacher {
+	return &windowsUpdatesCacher{
+		cacheStore:    cacheStore,
+		cacheInterval: cacheInterval,
+		slogger:       slogger.With("component", "windows_updates_cacher"),
+		interrupt:     make(chan struct{}, 1), // provide a buffer for the channel so that Interrupt can send to it and return immediately
+	}
+}
+
+func (w *windowsUpdatesCacher) Execute() (err error) {
+	cacheTicker := time.NewTicker(w.cacheInterval)
+	defer cacheTicker.Stop()
+
+	for {
+		select {
+		case <-cacheTicker.C:
+			// Since this query happens in the background and will not block auth, we can use
+			// a much longer timeout than we use for our tables.
+			var ctx context.Context
+			ctx, w.queryCancel = context.WithTimeout(context.Background(), 10*time.Minute)
+			if err := w.queryAndStoreData(ctx); err != nil {
+				w.slogger.Log(ctx, slog.LevelWarn,
```

Comment:
Maybe this is an error -- if we're seeing timeouts, we need to know asap, so we can change that number / behaviors

Reply:
Added: https://github.com/kolide/launcher/pull/2230/commits/2175f9ef9c52957735ec61ddd1b1384fd0cb44f5